File: easy-panel-backend/app/models/service.rb
class Service < ApplicationRecord
  belongs_to :employee, class_name: 'User', foreign_key: 'employee_id'

  validates :name, presence: true, uniqueness: { scope: :employee_id }
  validates :price, presence: true, numericality: { greater_than_or_equal_to: 0 }
  validates :duration, presence: true, numericality: { greater_than: 0 }

  scope :active, -> { where(is_active: true) }
  scope :for_employee, ->(employee_id) { where(employee_id: employee_id) }
end

File: easy-panel/src/types/service.ts
export interface Service {
  id: number;
  name: string;
  description: string;
  employee_id: number;
  price: string;
  duration: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  employee?: {
    id: number;
    name: string;
    email: string;
  };
}

export interface CreateServiceData {
  name: string;
  description: string;
  employee_id?: number;
  price: number;
  duration: number;
  is_active?: boolean;
}

export interface UpdateServiceData {
  name?: string;
  description?: string;
  employee_id?: number;
  price?: number;
  duration?: number;
  is_active?: boolean;
}

export interface ServiceFilters {
  employee_id?: number;
  is_active?: boolean;
}
